rest/main/main.go

```diff
@@ -218,3 +230,29 @@ func findAManagementIP() string {
 	glog.Warning("Could not find a management address!!")
 	return ""
 }
+
+// errorMsgPrefixes identifies the important error messages logged by
+// the standard http library that should not be missed
+var errorMsgPrefixes = [][]byte{
+	[]byte("http: Accept error"),
+	[]byte("http: panic"),
+	[]byte("http2: panic"),
+}
+
+// serverLog forwards the http library's logs to glog
+var serverLog = log.New(logWriter{}, "", 0)
+
+type logWriter struct{}
+
+func (logWriter) Write(p []byte) (int, error) {
+	for _, pfx := range errorMsgPrefixes {
+		if bytes.HasPrefix(p, pfx) {
+			glog.Warning(string(p))
```

Comment:
yes.. the http library uses log package. Hence actual source file & line was not getting printed earlier too. We also cannot use InfoDepth() since the depth value is not known here.

Reply:
Hmm. Did you actually try to use different values of depth in `InfoDepth()`? If the `http` library has a helper function that centralizes the place where the logs are written into `http.Server.ErrorLog` it is possible to configure it to show [the right place](https://go.dev/play/p/Seiw6ak4YNJ).

Reply:
Yes, i have considered all these techniques. Your example assumes that the passed log object directly in all places. But http server code uses internal wrapper functions to write the log messages (like [this](https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/net/http/server.go;l=3226) and [this](https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/net/http/h2_bundle.go;l=4359)). So, if I use the hardcoded depth as 3 or 4, i may not be getting the actual line which wrote this line. Also these file names and lines will not be useful -- they are from the http library code and not our code.

Reply:
I still believe that showing a location in a library is better than showing location in our code even if the fixed depth not always gives us the perfect result.

Reply:
Can you give a usecase where is better? I have seen instances where developers getting confused as they could not locate the file in the codebase. Also, if I understand correctly, the -vmodule filters do not consider this depth and always use caller's file. So, printing the library filename will also not help if someone wants to enable/disable these logs using v-level based on file name.

Reply:
I guess the difference in the approach to logging is resulting from having different requirements provided by ours and yours customers :smile: . Let's compromise an make the behavior user-selectable, say, via a command-line option.

Reply:
Command line option is an overkill. Not sure how customers can define this policy or care which file name gets printed in the log message. I was talking from developers & support engineer's perspective only. It is much easier to debug or involve right developers if they can locate the code that printed the log message.

Reply:
@tomek-US Can you please explain how showing a location in a library which has a wrapper is better than showing the actual place the logging is done from? While debugging where this log came from we will be showing the location of the wrapper which is not useful at all. Following is the snippet from your earlier comment.
"I still believe that showing a location in a library is better than showing location in our code even if the fixed depth not always gives us the perfect result."

Reply:
This problem can be solved by introducing a simple change:

```
	if bytes.HasPrefix(p, pfx) {
		if glog.V(3) {
			glog.WarningDepth(...)
		} else {
			glog.Warning(string(p))
		}
		return len(p), nil
	}
```

If you know that there is a wrapper adjusting the level by `+1` will result in providing location of the actual source of the message as demonstrated [here](https://go.dev/play/p/Seiw6ak4YNJ).

Reply:
It is impossible to know all the wrapper functions that are used not only in our code base but also the third party code and go lang standard code and keep adjusting the depth accordingly. Harding of depth like this is  not a wise approach in my opinion.